File: frontend/src/types/outfit.ts
import type { Garment } from './garment';

/** Outfit with garments populated. */
export interface Outfit {
  readonly id: string;
  readonly userId: string;
  readonly name: string;
  readonly previewImageUrl: string | null;
  readonly tags: readonly string[];
  readonly garments: readonly Garment[];
  readonly createdAt: string;
  readonly updatedAt: string;
}

/** Payload used to create a new outfit. */
export interface OutfitCreatePayload {
  readonly name: string;
  readonly garmentIds: readonly string[];
  readonly tags?: readonly string[];
}

File: backend/app/models/outfit.py
from __future__ import annotations

from datetime import datetime, timezone
from typing import Optional

from beanie import Document, Indexed
from pydantic import Field


class Outfit(Document):
    user_id: Indexed(str)
    name: str
    preview_image_url: Optional[str] = None
    garment_ids: list[str] = Field(default_factory=list)
    tags: list[str] = Field(default_factory=list)
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    updated_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))

    class Settings:
        name = 'outfits'

    def __repr__(self) -> str:
        return f"Outfit(id={self.id!r}, name={self.name!r})"
